packages/store/ts/config/v2/store.ts

```diff
@@ -42,10 +42,9 @@ export type resolveStore<store> = evaluate<{
   readonly tables: "tables" extends keyof store
     ? resolveTables<
         {
-          [key in keyof store["tables"]]: mergeIfUndefined<
-            store["tables"][key],
-            { namespace: get<store, "namespace"> }
-          >;
+          [key in namespacedTableKeys<store>]: key extends `${string}__${infer table}`
+            ? mergeIfUndefined<get<get<store, "tables">, table>, { namespace: get<store, "namespace">; name: table }>
+            : mergeIfUndefined<get<get<store, "tables">, key>, { namespace: get<store, "namespace">; name: key }>;
```

Comment:
why do we do this in reverse (namespace the table keys then extract the table name)? vs something like

```ts
{
  [tableName in keyof store['tables'] as resourceToLabel<store['namespace'], tableName>: ...
}
```

Reply:
true, this is the better approach here! had started from the world version where it's not possible because we change the shape of the object, but here it is possible